### ISSUE TITLE
How to manage DOTENV_PRIVATE_CI_KEY for multiple apps in monorepo?

### ISSUE THREAD

**Issue by vinnymac** [dotenvx/dotenvx#242]
## Summary

In the [documentation](https://dotenvx.com/docs/monorepos/turborepo#set-decryption-key) for `turborepo` I saw the following example:

```
$ npm run dev
web:dev: [dotenvx@0.43.2] injecting env (1) from .env
docs:dev: - ready started server on 0.0.0.0:3001, url: http://localhost:3001
web:dev: - ready started server on 0.0.0.0:3000, url: http://localhost:3000
```

And while this does work, it isn't entirely clear to me how one is supposed to use different environment variables for various apps/* or packages/* in a turborepo.

### Example

Let's say in the application above we had `apps/docs/.env.ci` and `apps/web/.env.ci`. What would be the recommended approach for say running `yarn turbo build` and having each command decrypt their respective environment variables via `DOTENV_PRIVATE_KEY_CI`?

In the example above, it looks like both `docs:dev` and `web:dev` commands share the same environment variables, but I believe splitting up environment variables across apps/packages is likely going to be the more common and secure approach for monorepos. What do you think?

Thanks

**Comment by motdotla** [comment]
This is a good question. Let me spend some time thinking on it. It seems to only be a challenge for ci. You're not blocked by production deploys with this, correct? because these would be deployed in different places - say 2 projects on vercel?

**Comment by bingtsingw** [comment]
I think it will work by just changing `DOTENV_PRIVATE_KEY_CI` manually.
For example,
in `docs/.env.ci`, change it to `DOTENV_PUBLIC_KEY_DOCS_CI`,
in `web/.env.ci`, change it to `DOTENV_PUBLIC_KEY_WEB_CI`,
and in `.env.keys`, set `DOTENV_PRIVATE_DOCS_KEY_CI` and `DOTENV_PRIVATE_KEY_WEB_CI`

**Comment by vinnymac** [comment]
@bingtsingw yea that’s what I am doing now, was just interested if a better approach was recommended.

If that’s the best practice, maybe we can update the docs with that info?

@motdotla yes, I am not blocked at all! Was just curious what y’all thought the recommended way to handle this was.

**Comment by motdotla** [comment]
not ready to add this as a best practice yet. what we might do is support comma-separated values:

```
DOTENV_PRIVATE_KEY_CI="abcd,efgh"
```

**Comment by motdotla** [comment]
@vinnymac can you give the comma separated setup a try. It looks like it should work as the code currently is - https://github.com/dotenvx/dotenvx/blob/main/src/lib/helpers/decryptValue.js#L10

**Comment by vinnymac** [comment]
That worked perfectly!

@motdotla thanks for the help here 🙇🏼

**Comment by motdotla** [comment]
great! i'll get that documented soon.

**Comment by psychobolt** [comment]
@motdotla Seems like we have to generate a separate key for each app inside a monorepo for setting environments. Is this necessary? 

Can we just use one key at the root that handle decrypt/encrypting? What would be useful is if we can reuse a existing key  path as mentioned in https://github.com/dotenvx/dotenvx/issues/209 . Concatenating the separate keys can be a chore for projects that don't really need separate encryption per sub-package basis.

**Comment by motdotla** [comment]
well, currently you technically don't have to manage separate. you could just put the same private key in each monorepo `.env.keys` file. 

or you could put your `.env.keys` in the root of your repo and run `dotenvx encrypt -f apps/app1/.env`

i kind of like that idea but servicing all the various custom needs for monorepos gets tricky. everyone uses them differently.

**Comment by motdotla** [comment]
> Without duplicating the initially created .env.keys into the subdirectories, this approach would generate a new private/public key pair, which I’d prefer to avoid for simplicity.

are you sure? It shouldn't.

```
mkdir -p apps/app1
touch .env
touch apps/app1/.env
dotenvx set HELLO World
dotenvx set HELLO App1 -f apps/app1/.env
```

oops, it doesn't! let me look into this. this i think is a bug and would make monorepos very difficult

**Comment by motdotla** [comment]
actually, going down that path is going to be a boondoggle - which is probably why I didn't implement it like that originally.

instead, I just need to let users control the location of the keys file with a `--env-keys-file, -fk` flag.

effort started here: https://github.com/dotenvx/dotenvx/pull/486

**Comment by bingtsingw** [comment]
Looking forward to `-fk` flag

**Comment by motdotla** [comment]
released in `1.30.0`. let me know if you run into any issues with it everyone.

https://github.com/dotenvx/dotenvx/issues/209
